cmd/dep/ensure.go

```diff
@@ -779,13 +786,47 @@ func getProjectConstraint(arg string, sm gps.SourceManager) (gps.ProjectConstrai
 	}
 
 	pi := gps.ProjectIdentifier{ProjectRoot: pr, Source: source}
+
+	// check to see whether our versionStr contains what looks to be a semver
+	// tag -- this is so we can make sure it's a full version and not a range
+	if len(versionStr) > 0 {
+		vs := lvRegexp.FindString(versionStr)
+
+		switch true {
+		case len(vs) == 0:
+			// there is no semver value present -- do nothing
+		case versionStr == vs:
+			// the version is an absolute version -- prepend it with '='
+			versionStr = "=" + versionStr
+		case len(versionStr) > len(vs):
+			// the version string contains more than the semver string
+			// try to make sure it's not a range of versions
+			if !checkVersionString(versionStr) {
+				return emptyPC, "", errors.Errorf("version string %q is not valid: must be an absolute semantic version", versionStr)
+			}
+		}
+	}
+
 	c, err := sm.InferConstraint(versionStr, pi)
 	if err != nil {
 		return emptyPC, "", err
 	}
 	return gps.ProjectConstraint{Ident: pi, Constraint: c}, arg, nil
 }
 
+func checkVersionString(full string) bool {
```

Comment:
I could probably make this function name a bit more descriptive. `checkNotVersionRange` might be better. Thoughts?

Reply:
I'd call it `isVersionRange` or `isSemVerRange`.